src/components/TaskItem/TaskItem.tsx

```diff
@@ -1,19 +1,34 @@
+import { useState } from "react";
 import type { Task } from "../../types";
 import "./TaskItem.css";
 import { Button } from "../Button/Button";
-import { EditIcon, DeleteIcon } from "../Icons/Icons";
+import { EditIcon, DeleteIcon, SaveIcon } from "../Icons/Icons";
+import { NewTaskInput } from "../NewTaskInput/NewTaskInput";
 
 type Props = {
   task: Task;
   onComplete: () => void;
   onDelete: () => void;
+  onUpdate: (updatedTask: Task) => void;
 };
 
-const handleEditTask = () => {
-  console.log("Edited!");
-};
+const TaskItem = ({ task, onComplete, onDelete, onUpdate }: Props) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [taskText, setTaskText] = useState(task.title);
+
+  const handleEditTask = () => {
+    setIsEditing(true);
+  };
+
+  const handleChange = (value: string) => {
+    setTaskText(value);
+  };
+
+  const handleSave = () => {
+    setIsEditing(false);
+    onUpdate({ ...task, title: taskText });
```

Comment:
в дополнение к идее про передачу только вэлью и айди
```suggestion
    onUpdate(task.id, taskText);
```